### DISCUSSION TITLE
783

### DISCUSSION THREAD

**Discussion by giscus** [jojoldu/blog-comments#954] [Comments]
# 783

마틴 파울러의 블로그를 가보면 Leonard Richardson이 제안한 HTTP API 성숙도 모델 (Richardson Maturity Model - RMM)을 소개한다.여기서는 다음과 같이 4단계를 설명한다.레벨 0: 하나의 URI를 정의하고 모든 작업은 이 URI에 대한 POST 요청레벨 1: 개별 리소스에 대해 별도의 URI를 만든다.레벨 2: HTTP 메서드를 사용하여 리소스에 대한 작업을 정의레벨 3: 하이퍼미디어(HATEOAS 등)를 사용Roy Fielding 의 정의에 따르면 레벨 3이 진정한 RESTful API에 해당한다.대부분의 RESTful API는 진짜 RESTful API는 아니며 보통 레벨2 부근에 해당한다.아래는 마틴 파울러 블로그에 나와있는 성숙도 레벨이 무엇을 의미하는지..

https://jojoldu.tistory.com/783

**Comment by jws1837** [comment]
이건 너무 좋네요.  많은 고민이 느껴집니다!

**Comment by intellijh** [comment]
항상 신세 많이 지고 있습니다

**Comment by mag123c** [comment]
작성했던, 앞으로 작성해 나갈 API URI들에 대해 저 또한 깊게 생각해보게 되는 너무 좋은 글이에요!!
늘 좋은 인사이트를 제공해주셔서 감사합니다!

(오타 제보드립니다!)
(성숙도 레벨이 무엇을 의미하는지에 대햔 섹션 : "대햔")

**Comment by mushroom528** [comment]
잘 읽었습니다! 감사합니다
path에 id를 추가하는경우 위치를 중간이 아닌 마지막에 추가하는 경우가 있는데 어떤 것을 더 선호하시나요?
POST /orders/{orderId}/cancel(중간에 위치하는 경우)
POST /orders/cancel/{orderId}(끝에 위치하는 경우)
